sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -993,6 +993,30 @@ object functions {
     Lead(e.expr, Literal(offset), Literal(defaultValue))
   }
 
+  /**
+   * Window function: returns the value that is the `offset`th row of the window frame
+   * (counting from 1), and `null` if the size of window frame is less than `offset` rows.
+   *
+   * This is equivalent to the nth_value function in SQL.
+   *
+   * @group window_funcs
+   * @since 3.0.0
+   */
+  def nth_value(columnName: String, offset: Int): Column = {
```

Comment:
@beliefer, how is it different from `leg`?

Reply:
They are existing two differences.
The first is:
The offset of `lag` based on the current row in the window. 
The offset of `nth_value` based on the first row in the window. 
The second is.
The offset of `lag` increases from back to front.
The offset of `nth_value` increases from front to back.